### PR TITLE
VideoBackends: Remove the option to disable fast depth.

### PR DESCRIPTION
**Pull request by Armada651**
Fast depth is now accurate on D3D and on OGL with drivers that support `GL_ARB_clip_control`. Now that accuracy is no longer an argument we should eliminate the fast depth option and keep it enabled.

Drivers that do not yet support `GL_ARB_clip_control` will calculate the depth value in the pixel shader. This makes testing and debugging of depth issues easier, since we won't have to worry about depth bug reports caused by an inaccurate fast depth. Any modern driver supports the extension this includes recent versions of Nvidia, AMD and Mesa.

### CODE REVIEW

**Comment by skidau** [comment]
This works for me. Thanks.

File: Source/Core/VideoCommon/VertexShaderGen.cpp [hunk 1]
```diff
@@ -377,32 +377,15 @@ static inline void GenerateVertexShader(T& out, u32 components, API_TYPE api_typ
...
 	}
-	else if (api_type == API_D3D)
-	{
-		out.Write("o.pos.z = -o.pos.z;\n");
-	}
-	else // OGL
-	{
-		// this results in a scale from -1..0 to -1..1 after perspective
-		// divide
-		out.Write("o.pos.z = o.pos.z * -2.0 - o.pos.w;\n");
```
Comment (degasus) [inline Source/Core/VideoCommon/VertexShaderGen.cpp]: We still need this path for zcomploc

**Comment by degasus** [comment]
I'm not sure if we want to include the binary file docs/Depth/dolphin-depth.pdf within the repo.

**Comment by Armada651** [comment]
We have other PDFs in there, so I thought it was alright.

File: Source/Core/VideoCommon/VertexShaderGen.cpp (hunk 1 above)
Comment (Armada651) [inline Source/Core/VideoCommon/VertexShaderGen.cpp]: Can we still remove the `- o.pos.w` though? That shouldn't affect the depth test, but would remove the rounding error.

**Comment by neobrain** [comment]
I still think the PDF is way too cryptic for anyone other than the people who lengthily discussed the matter already to be useful. Is there no way you could clear it up by at least adding one or two sentences of context?

**Comment by Armada651** [comment]
Will do as soon as I we've fixed the remaining differences between slow depth and fast depth.

**Comment by shuffle2** [comment]
rebase plz

**Comment by Armada651** [comment]
This PR has been reopened as PR #2460.

**Pull request closed without merging**